cmd/virt-launcher/sock-connector

```diff
@@ -1,4 +1,28 @@
 #!/bin/bash
-stty -echo
+
 SOCKET=$1
+function serial_cleanup() {
+	local pid_file=${SOCKET}.pid
+	local file_name=$(basename $SOCKET)
+
+	# if this is a serial connection, see if there is a previous
+	# connection that must be cleaned up before starting a new one.
+	if [[ $file_name =~ .*serial.* ]]; then
+		local pid=$(cat ${SOCKET}.pid 2>/dev/null)
```

Comment:
Using "$pid_file" here might be more readable than "${SOCKET}.pid".

Reply:
yup it would be. i'll update this